src/config.rs

```diff
@@ -0,0 +1,233 @@
+use std::{
+    fs,
+    env,
+    path::Path,
+    sync::{Arc, Mutex},
+};
+use std::fs::File;
+use std::io::Write;
+use std::process::exit;
+use regex::Regex;
+use serde::Deserialize;
+use toml::value;
+
+/// The response that is injected into a panic, if the config file was configured falsely
+const PANIC_RESPONSE: &str = "Please create a config file yourself or try setting the environment CONFIG_FILE to valid file location!";
+
+/// The versions of the config this program is compatible with
+const COMPATIBLE_VERSIONS: [(u32, u32); 1] = [
+    (0, 2),
+];
+
+lazy_static! {
+    ///
+    /// The global, thread safe configuration of all of this bot.
+    ///
+    /// Examples:
+    /// ```
+    ///
+    /// let config_arc = Arc::clone(&CONFIG);
+    /// let mut config_lock = config_arc.lock();
+    /// while config_lock.is_err() {
```

Comment:
A error in locking means the lock has been poisoned and can't be unlocked again.
This should panic instead of retrying!